sotopia/database/evaluation_dimensions.py

```diff
@@ -0,0 +1,232 @@
+            dimension = CustomEvaluationDimension.get(dimension_id)
+            range_validator = EvaluationDimensionGenerator.create_range_validator(
+                dimension.range_low, dimension.range_high
+            )
+            field_type = Annotated[Tuple[str, int], range_validator]
+
+            fields[dimension.name] = (
+                field_type,
+                Field(..., description=dimension.description),
+            )
+
+        return create_model(
+            "CustomEvaluationDimensionModel",
+            __base__=BaseModel,
+            **fields,
+        )
+
+    @staticmethod
+    def generate_dimension_model_from_dict(
+        dimensions: list[dict[str, Union[str, int]]],
+    ) -> Type[BaseModel]:
+        fields = {}
+        for dimension_dict in dimensions:
+            dimension = CustomEvaluationDimension(**dimension_dict)
+            range_validator = EvaluationDimensionGenerator.create_range_validator(
+                dimension.range_low, dimension.range_high
+            )
+            field_type = Annotated[Tuple[str, int], range_validator]
+
+            fields[dimension.name] = (
+                field_type,
+                Field(..., description=dimension.description),
+            )
+
+        return create_model(
+            "CustomEvaluationDimensionModel",
+            __base__=BaseModel,
+            **fields,
+        )
+
+    @staticmethod
+    def generate_dimension_model_from_name(
+        dimension_names: list[str],
+    ) -> Type[BaseModel]:
+        # Migrator().run()
+        from sotopia.database import EpisodeLog
+
+        print(len(list(EpisodeLog.all_pks())))
+        print(list(CustomEvaluationDimension.all_pks()))
+        print(
+            [
+                CustomEvaluationDimension.get(pk)
+                for pk in CustomEvaluationDimension.all_pks()
+            ]
+        )
+
+        # print(CustomEvaluationDimension.find(CustomEvaluationDimension.name == dimension["name"]).all()[0], "already exists")
+        fields = {}
+        for dimension_name in dimension_names:
+            dimension = CustomEvaluationDimension.find(
+                CustomEvaluationDimension.name == dimension_name
+            ).all()
+            assert (
+                len(dimension) == 1
+            ), f"Expected 1 dimension for {dimension_name}, but found {len(dimension)}"
+            dimension = dimension[0]
+            range_validator = EvaluationDimensionGenerator.create_range_validator(
+                dimension.range_low, dimension.range_high
+            )
+            field_type = Annotated[Tuple[str, int], range_validator]
+
+            fields[dimension.name] = (
+                field_type,
+                Field(..., description=dimension.description),
+            )
+
+        return create_model(
+            "CustomEvaluationDimensionModel",
+            __base__=BaseModel,
+            **fields,
+        )
+
+
+def generate_sotopia_dimensions() -> Type[BaseModel]:
```

Comment:
we don't need to generate these over and over again? I would suggest move these to the doc